components/data_provider_json/tests/test_file_io.rs

```diff
@@ -0,0 +1,35 @@
+use icu_data_provider_json;
+
+use std::borrow::Cow;
+
+use std::fs::File;
+use std::io::BufReader;
+
+use icu_data_provider as datap;
+use icu_data_provider::DataProvider;
+use icu_data_provider_json::JsonDataProvider;
+
+#[test]
+fn test_read_json() {
+    let file = File::open("tests/testdata/all.json").unwrap();
+    let reader = BufReader::new(file);
+    let json_data_provider = JsonDataProvider::from_reader(reader).unwrap();
+    println!("{:?}", json_data_provider); // Coverage for Debug trait
```

Comment:
It'd be better to actually test - https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=752a379a1e03fe77995ff0e00649f032

Reply:
Do we want to assert the format for a Debug trait?  If the message changes, that shouldn't break our test I think.

Reply:
Follow-up: #118